### ISSUE TITLE
Unknown field 'metafield' on type 'ShopifyProduct'

### ISSUE THREAD

**Issue by hanshank** [gatsbyjs/gatsby#25291]
## Description

I've been using gatsby-source-shopify and it's been working just fine. However, after following the guides for exposing/whitelisting metafields to the Storefront API, I still get the following error when trying to query for the metafield like so:

```
query($handle: String!) {
    shopifyProduct(handle: { eq: $handle }) {
...
      metafield(namespace: "custom_fields", key: "seo_title") {
        value
      }
...
}
```

### Steps to reproduce

1. fire up `gatsby build` or `gatsby develop` will both cause this error: '- Unknown field 'metafield' on type 'ShopifyProduct'.' on an individual product query.

### Expected result

Expected to get the metafield value.

### Actual result

I got this error instead:

Unknown field 'metafield' on type 'ShopifyProduct'

### Environment

  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.14.1 - ~/.nvm/versions/node/v12.14.1/bin/node
    Yarn: 1.22.4 - ~/.nvm/versions/node/v12.14.1/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.14.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 83.0.4103.116
    Firefox: 76.0.1
    Safari: 13.0.5
  npmPackages:
    gatsby: ^2.17.4 => 2.17.4
    gatsby-image: ^2.2.29 => 2.2.29
    gatsby-plugin-google-analytics: ^2.0.13 => 2.1.23
    gatsby-plugin-layout: ^1.0.11 => 1.1.13
    gatsby-plugin-manifest: ^2.2.23 => 2.2.23
    gatsby-plugin-netlify-cms: ^4.3.0 => 4.3.0
    gatsby-plugin-netlify-identity-widget: ^1.1.1 => 1.1.1
    gatsby-plugin-offline: ^3.0.16 => 3.0.16
    gatsby-plugin-react-helmet: ^3.1.13 => 3.1.13
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0
    gatsby-plugin-root-import: ^2.0.5 => 2.0.5
    gatsby-plugin-sharp: ^2.2.32 => 2.2.32
    gatsby-remark-images: ^3.3.11 => 3.3.11
    gatsby-remark-relative-images: ^0.3.0 => 0.3.0
    gatsby-source-filesystem: ^2.1.33 => 2.1.33
    gatsby-source-shopify: ^3.2.14 => 3.2.14
    gatsby-transformer-json: ^2.3.3 => 2.3.3
    gatsby-transformer-remark: ^2.8.0 => 2.8.0
    gatsby-transformer-sharp: ^2.3.0 => 2.3.0
  npmGlobalPackages:
    gatsby-cli: 2.12.1

**Comment by smthomas** [comment]
@hanshank Thanks for the issue report. This is likely because that field is not set as an available field in the queries.js file in `gatsby-source-shopify`.

We recently merged in #23922 which allows you to customize the queries that are sent to Shopify (which allows you to add fields that can be accessed with GraphQL within Gatsby). If you check the README of the gatsby-source-shopify plugin you will see there is additional documentation on how to use this.

https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-shopify#shopifyqueries-defaults

I think if you add the `metafield` field to a as a custom query in your gatsby-config.js it might work.

**Comment by hanshank** [comment]
Thanks for getting back to me @smthomas, I appreciate it. 

This is a cool feature and I will definitely keep in mind for future projects. I already solved this one accessing the Shopify Graphql Storefront API outside of the gatsby-source-shopify plugin.

Thanks again!
